main.js

```diff
@@ -387,7 +416,12 @@ function renderApp(state) {
     $view.appendChild(renderConfirmation())
   }
   else {
-    $view.appendChild(renderGrid(state.catalog.items))
+    if (state.sort !== null) {
+      $view.appendChild(renderGrid(state.sortedCatalog))
```

Comment:
Instead of keeping a sorted copy around, create a sorted copy right at the last second and pass it to your `renderGrid` function.